components/locale/src/extensions/unicode/attribute.rs

```diff
@@ -0,0 +1,91 @@
+use std::ops::RangeInclusive;
+use std::str::FromStr;
+
+use crate::parser::errors::ParserError;
+use tinystr::TinyStr8;
+
+#[derive(Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord, Copy)]
+pub struct Attribute(TinyStr8);
+
+const ATTR_LENGTH: RangeInclusive<usize> = 3..=8;
+
+impl Attribute {
```

Comment:
It took me a while to find what attributes are in locale spec. It seems that they exist only as placeholders:

> no attribute subtags are defined by the current CLDR specification

Nit: Maybe add it to the doc text, since it's not clear from sample code what they do (which is expected, since nobody uses them).